src/beacons/use-beacons.ts

```diff
@@ -200,7 +237,7 @@ const allowedPermissionForKettle = async () => {
   if (permissionStatuses.bluetooth) {
     kettleModulesArray.push(KettleModules.BLUETOOTH);
   }
-  if (permissionStatuses.location) {
+  if (permissionStatuses.locationWhenInUse || permissionStatuses.locationAlways) {
```

Comment:
```suggestion
  if (permissionStatuses.locationAlways) {
```
Since locationWhenInUse is not a specific beacons permission, this could cause `permissions.length > 0` to be true even before the relevant onboarding screen has been shown.